index.js

```diff
@@ -1,26 +1,120 @@
 // 1.Fibonacci
 
-// function fibo(n) {
-//   return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
-// }
-// fibo(5);
-// fibo(6);
-// fibo(7);
+function fibo(n) {
+  return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
+}
+fibo(5);
+fibo(6);
+fibo(7);
 
 // 2. Implementing with reduce
 
 //concat;
+let arr = [1, 2];
+let arr1 = [3, 4];
+
+const concatFn = (...array) => {
+  return array.reduce((array, newEl) => [...array, ...newEl]);
+};
+
+concatFn(arr, arr1);
+
+//filter and map;
+
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+
+let adult = friends.reduce(function (newArr, person) {
```

Comment:
this is not an implementation of filter function
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

your condition is hardcoded, filter function must be generic and accept any condition